server/s3_list_object_versions_action.cc

```diff
@@ -115,7 +115,29 @@ void S3ListObjectVersionsAction::validate_request() {
   request_key_marker = request->get_query_string_value("key-marker");
   s3_log(S3_LOG_DEBUG, request_id, "request_key_marker = %s\n",
          request_key_marker.c_str());
-  last_key = request_key_marker;  // As requested by user
+  if (request->has_query_param_key("version-id-marker")) {
+    request_version_id_marker =
+        request->get_query_string_value("version-id-marker");
+    s3_log(S3_LOG_INFO, request_id, "request_version_id_marker = %s\n",
+           request_version_id_marker.c_str());
+    if (request_version_id_marker.empty()) {
+      s3_log(S3_LOG_DEBUG, request_id, "version-id-marker is empty\n");
+      // TODO: Add invalid argument details to the response.
+      set_s3_error("InvalidArgument");
```

Comment:
```suggestion
      set_s3_error("InvalidArgument");
      set_s3_error_message("A version-id marker cannot be empty.");
      set_invalid_argument("version-id-marker", request_version_id_marker);
```

Here's the result from AWS:

```txt
$ aws s3api list-object-versions --bucket mybucket --version-id-marker '' --key-marker shell.nix

An error occurred (InvalidArgument) when calling the ListObjectVersions operation: A version-id marker cannot be empty.
```

Debug output:

```xml
<?xml version="1.0" encoding="UTF-8"?>\n
<Error>
    <Code>InvalidArgument</Code>
    <Message>A version-id marker cannot be empty.</Message>
    <ArgumentName>version-id-marker</ArgumentName>
    <ArgumentValue></ArgumentValue>
    <RequestId>VH8P4E93T962G3WX</RequestId>
    <HostId>iRlqrKmf+/CcVbcE6qjncj4SoHvZajRb9wq1n6CL7cvpXtnJMTealm20ZMwHuzMpaXjwenZ+c+k=</HostId>
</Error>
```

Reply:
Done